modules/drivers/redshift/src/metabase/driver/redshift.clj

```diff
@@ -26,6 +27,18 @@
 ;;; |                                             metabase.driver impls                                              |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+;; custom Redshift type handling
+
+(def ^:private database-type->base-type
+  (sql-jdbc.sync/pattern-based-database-type->base-type
+    [
+     [#"(?i)CHARACTER VARYING" :type/Text]       ; Redshift uses CHARACTER VARYING (N) as a synonym for VARCHAR(N)
+     [#"(?i)NUMERIC"           :type/Decimal]])) ; and also has a NUMERIC(P,S) type, which is the same as DECIMAL(P,S)
+
+(defmethod sql-jdbc.sync/database-type->base-type :redshift
+  [_ column-type]
+  (database-type->base-type column-type))
+
```

Comment:
you should probably put these in the `metabase.driver.sql-jdbc impls` section below instead of here with other `metabase.driver` method impls

Reply:
Moved below.